framework/core/src/Http/AccessToken.php

```diff
@@ -112,7 +112,8 @@ public function touch(ServerRequestInterface $request = null)
             // We truncate user agent so it fits in the database column
             // The length is hard-coded as the column length
             // It seems like MySQL or Laravel already truncates values, but we'll play safe and do it ourselves
-            $this->last_user_agent = substr(Arr::get($request->getServerParams(), 'HTTP_USER_AGENT'), 0, 255);
+            $agent = Arr::get($request->getServerParams(), 'HTTP_USER_AGENT');
+            $this->last_user_agent = $agent !== null ? substr($agent, 0, 255) : null;
```

Comment:
This results in different behaviour. Passing `null` to `substr()` will default to an empty string, but this falls back to null.

```suggestion
            $this->last_user_agent = ? substr($agent ?? "", 0, 255);
```